File: flatten-multidimensional-array/flattenMultidimensionalArray.js
'use strict'

const flattenMultidimensionalArray = array => {
  const resultantArray = []

  for (const item of array) {
    if (Array.isArray(item)) {
      resultantArray.push(...flattenMultidimensionalArray(item))
    } else {
      resultantArray.push(item)
    }
  }

  return resultantArray
}

module.exports = flattenMultidimensionalArray

File: flatten-multidimensional-array/flatten_multidimensional_array.rb
def flatten_multidimensional_array(array)
  resultant_array = []

  array.each do |item|
    if item.is_a?(Array)
      resultant_array.concat(flatten_multidimensional_array(item))
    else
      resultant_array.push(item)
    end
  end

  resultant_array
end
